src/values.md

```diff
@@ -0,0 +1,186 @@
+
+> [!WARN]
+> On `little` endian, the order of bytes used to decode an integer type is the same as the natural order of a `u8` array - that is, the `m` value corresponds with the `m` index into a same-sized `u8` array. On `big` endian, however, the order is the opposite of this order - that is, the `m` value corresponds with the `size_of::<T>() - m` index in that array.
+
+r[value.primitive.signed-repr]
+A value `i` of a signed integer type with width `N` is represented the same as the corresponding value of the unsigned counterpart type which is congruent modulo `2^N`.
+
+r[value.primitive.char]
+Each value of type `char` is a Unicode Scalar Value, between `U+0000` and `U+10FFFF` (excluding the surrogate range `U+D800` through `U+DFFF`).
+
+r[value.primitive.char-repr]
+The representation of type `char` is the same as the representation of the `u32` corresponding to the Code Point Number encoding by the `char`.
+
+r[value.primitive.bool]
+The two values of type `bool` are `true` and `false`. The representation of `true` is an initialized byte with value `0x01`, and the representation of `false` is an initialized  byte with value `0x00`. Neither value is represented with a pointer fragment.
+
+r[value.primitive.float]
+A floating-point value consists of either a rational number, which is within the range and precision dictated by the type, an infinity, or a NaN value.
+
+r[value.primitive.float-repr]
+A floating-point value is represented the same as a value of the unsigned integer type with the same width given by its [IEEE 754-2019] encoding.
+
+r[value.primitive.float-format]
+The [IEEE 754-2019] `binary32` format is used for `f32`, and the `binary64` format is used for `f64`.
+
+[IEEE 754-2019]: https://ieeexplore.ieee.org/document/8766229
+
+## Pointer Value
+
+r[value.pointer]
+
+r[value.pointer.thin]
+Each thin pointer consists of an address and an optional provenance. The address refers to which byte the pointer points to. The provenance refers to which bytes the pointer is allowed to access, and the allocation those bytes are within.
+
+> [!NOTE]
+> A pointer that does not have a provenance may be called an invalid or dangling pointer.
+
+r[value.pointer.thin-repr]
+The representation of a value of a thin pointer is a sequence of initialized bytes with `u8` values given by the representation of its address as a value of type `usize`, and pointer fragments corresponding to its provenance, if present.
+
+r[value.pointer.thin-ref]
+A thin reference to `T` consists of a non-null, well aligned address, and provenance for `size_of::<T>()` bytes starting from that address. The representation of a thin reference to `T` is the same as the pointer with the same address and provenance.
+
+> [!NOTE]
+> This is true for both shared and mutable references. There are additional constraints enforced by the aliasing model.
+
+r[value.pointer.fat]
+A fat pointer or reference consists of a data pointer or reference, and a pointee-specific metadata value.
```

Comment:
The Rust reference currently uses the term "wide" in many places, e.g. "wide reference", "wide pointer". The compiler does as well, such as in the lint `ambiguous_wide_pointer_comparisons` and in many places in the source code. We should be consistent, and use `valid.pointer.wide` here, unless there's some strong reason to do otherwise.

(In addition to consistency, "wide" also seems like more inclusive terminology.)

Reply:
As Josh noted, this was decided by FCP here:

- https://github.com/rust-lang/rust/pull/117758#issuecomment-1832381065